updated for ninja 1.11.0 on gentoo
patch updated from 1.8.2-1 to ninja 1.9.0-1

modified to actually accept -c instead of fail
also don't return, but break instead! else other flags are ignored!

Originally from: (as PR 1268 on github):
From 7d27df3873593f573515f3133ba7766d8d4fb262 Mon Sep 17 00:00:00 2001
From: Alexander Timin <altimin@chromium.org>
Date: Wed, 12 Apr 2017 14:11:48 +0100
Subject: [PATCH] Add -c flag to force color output.

---
 src/build.cc | 3 +--
 src/build.h  | 5 ++++-
 src/ninja.cc | 6 +++++-
 3 files changed, 10 insertions(+), 4 deletions(-)

diff -upr ninja-1.9.0/src/build.h ninja-1.9.0.mod/src/build.h
--- ninja-1.9.0/src/build.h	2019-01-30 19:58:59.000000000 +0100
+++ ninja-1.9.0.mod/src/build.h	2019-03-04 19:34:38.895254540 +0100
@@ -138,7 +138,8 @@ struct CommandRunner {
 /// Options (e.g. verbosity, parallelism) passed to a build.
 struct BuildConfig {
   BuildConfig() : verbosity(NORMAL), dry_run(false), parallelism(1),
-                  failures_allowed(1), max_load_average(-0.0f) {}
+                  failures_allowed(1), max_load_average(-0.0f),
+                  force_color_output(false) {}
 
   enum Verbosity {
     NORMAL,
@@ -153,6 +154,8 @@ struct BuildConfig {
   /// means that we do not have any limit.
   double max_load_average;
   DepfileParserOptions depfile_parser_options;
+  // Do not strip color marks even when writing to a non-terminal.
+  bool force_color_output;
 };
 
 /// Builder wraps the build process: starting commands, updating status.
diff -upr ninja-1.9.0/src/ninja.cc ninja-1.9.0.mod/src/ninja.cc
--- ninja-1.9.0/src/ninja.cc	2019-01-30 19:58:59.000000000 +0100
+++ ninja-1.9.0.mod/src/ninja.cc	2019-03-04 19:33:35.095056617 +0100
@@ -219,7 +219,8 @@ void Usage(const BuildConfig& config) {
 "  -d MODE  enable debugging (use '-d list' to list modes)\n"
 "  -t TOOL  run a subtool (use '-t list' to list subtools)\n"
 "    terminates toplevel options; further flags are passed to the tool\n"
-"  -w FLAG  adjust warnings (use '-w list' to list warnings)\n",
+"  -w FLAG  adjust warnings (use '-w list' to list warnings)\n"
+"  -c       force using color output\n",
           kNinjaVersion, config.parallelism);
 }
 
@@ -1123,7 +1124,7 @@ int ReadFlags(int* argc, char*** argv,
 
   int opt;
   while (!options->tool &&
-         (opt = getopt_long(*argc, *argv, "d:f:j:k:l:nt:vw:C:h", kLongOptions,
+         (opt = getopt_long(*argc, *argv, "d:f:j:k:l:nct:vw:C:h", kLongOptions,
                             NULL)) != -1) {
     switch (opt) {
       case 'd':
@@ -1185,6 +1186,9 @@ int ReadFlags(int* argc, char*** argv,
       case OPT_VERSION:
         printf("%s\n", kNinjaVersion);
         return 0;
+      case 'c':
+        config->force_color_output = true;
+        break;
       case 'h':
       default:
         Usage(*config);
Index: /var/tmp/portage/dev-util/ninja-1.11.0/work/ninja-1.11.0/src/status.cc
===================================================================
--- ninja-1.11.0/src/status.cc
+++ ninja-1.11.0/src/status.cc
@@ -102,7 +102,7 @@ void StatusPrinter::BuildEdgeFinished(Ed
     // only a few hundred available on some systems, and ninja can launch
     // thousands of parallel compile commands.)
     string final_output;
-    if (!printer_.supports_color())
+    if (!printer_.supports_color() && !config_.force_color_output)
       final_output = StripAnsiEscapeCodes(output);
     else
       final_output = output;
